File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        int m = 0;
        int length = 1;
        set<int> set(nums.begin(), nums.end());
        for(int num : set){
            if(set.find(num + 1) != set.end()){
                length++;
            }
            else{
                m = max(m, length);
                length = 1;
            }
        }
        m = max(m, length);
        return m;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int max = 0;
        int size = set.size();
        for(int num : set){
            if (!set.contains(num - 1)) {
                int start = num;
                int count = 0;
                while(set.contains(start+1)){
                    start++;
                    count++;
                }
                max = Math.max(max, count + 1);
            }
        }
        return max;
    }
}